tests/test_matrix_utils_connect_credentials.py

```
@@ -0,0 +1,308 @@
+        patch("mmrelay.matrix_utils.logger") as mock_logger,
+    ):
+        client = await connect_matrix(config)
+
+    assert client is mock_client
+    mock_logger.info.assert_any_call(
+        "NOTE: Ignoring Matrix login details in config.yaml in favor of credentials.json"
+    )
+
+
+@pytest.mark.asyncio
+async def test_connect_matrix_auto_login_load_credentials_failure(monkeypatch):
```

Ruff (0.15.9)

[warning] Unused function argument: `monkeypatch`

(ARG001)